lib/rules-generator.js

```diff
@@ -218,25 +251,34 @@ util.methods(Generator, {
     var i;
```

Comment:
would be more functional if it returned a new instace of the enclosing class rather than mutating its internal state in a hard to reason about way.